Add readiness probe and fix readme (#63)
rancher/k3k: README.md

[![asciicast](https://asciinema.org/a/eYlc3dsL2pfP2B50i3Ea8MJJp.svg)](https://asciinema.org/a/eYlc3dsL2pfP2B50i3Ea8MJJp)

## Architecture

K3K consists of a controller and a cli tool, the controller can be deployed via a helm chart and the cli can be downloaded from the releases page.

### Controller

The K3K controller will watch a CRD called `clusters.k3k.io` Once found, the controller will create a separate namespace and it will create a K3S cluster as specified in the spec of the object.

> Each server and agent is created as a separate pod that runs in that namespace.

Reviewer: ```suggestion
Each server and agent is created as a separate pod that runs in the new namespace.
```

### CLI

> The CLI will provide a quick way to create K3K clusters using simple flags. The CLI automatically exposes the K3K clusters so it's accessible via a kubeconfig.

Reviewer: ```suggestion
The CLI provides a quick and easy way to create K3K clusters using simple flags, and automatically exposes the K3K clusters so it's accessible via a kubeconfig.
```

## Features

### Isolation

> Each cluster runs in a sperate namespace that can be isolated via netowrk policies and RBAC rules, clusters also run in a sperate network namespace with flannel as the backend CNI, finally each cluster has a separate datastore which can be persisted.

Reviewer: ```suggestion
Each cluster runs in a sperate namespace that can be isolated via netowrk policies and RBAC rules, clusters also run in a sperate network namespace with flannel as the backend CNI. Finally, each cluster has a separate datastore which can be persisted.
```

Reviewer: Can we add a bit here about how the datastore can be persisted? I think there'd be value in highlighting how users can have the benefit of Kine, etc.

Author: I think it still needs some code modifications, since we are defaulting now to the etcd backend, it can be done but I opted not to mention it until its actually available

> ### Protability and Customization

Reviewer: ```suggestion
### Portability and Customization
```

> Cluster object is considered the template of the cluster where you can re-use to spin up multiple clusters in matter of seconds.

Reviewer: ```suggestion
The "Cluster" object is considered the template of the cluster that you can re-use to spin up multiple clusters in a matter of seconds.
```

> K3K clusters use k3s internally and leverage all options that can be passed to k3s, each cluster is exposed to the host cluster via NodePort, LoadBalancers, and Ingresses.

Reviewer: ```suggestion
K3K clusters use K3S internally and leverage all options that can be passed to K3S. Each cluster is exposed to the host cluster via NodePort, LoadBalancers, and Ingresses.
```
